src/System.Net.Requests/tests/HttpRequestResponseSerializationTest.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization.Formatters.Tests;
+using Xunit;
+
+namespace System.Net.HttpRequest.Tests
+{
+    public partial class HttpRequestResponseSerialization
+    {
+        public readonly static object[][] EchoServers = System.Net.Test.Common.Configuration.Http.EchoServers;
+
+        [OuterLoop]
+        [Theory, MemberData(nameof(EchoServers))]
+        public void SerializeDeserialize_Roundtrips(Uri remoteServer)
+        {       
+            HttpWebRequest request = WebRequest.CreateHttp(remoteServer);
+            request.KeepAlive = true;
+            request.CookieContainer = new CookieContainer(10);
+            HttpWebRequest clone = BinaryFormatterHelpers.Clone(request);
+            Assert.NotSame(request, clone);
```

Comment:
@karelz @stephentoub @davidsh This is more complicated. I'll submit by PR as-is without serialization working and open a new tracking issue.

All internal APIs rely on `_httpResponseMessage` which is private. Serialization won't copy/populate this field and so most of the properties can't be used (throw `System.ObjectDisposedException`).

There's an additional issue with `ContentLength` not having a setter (although the ported code is similar to NetFX). That may be a limitation of the Core serializer.